src/eventProcessor/GoogleAnalyticsEventProcessor.js

```diff
@@ -0,0 +1,129 @@
+goog.module('measurementLibrary.eventProcessor.GoogleAnalyticsEventProcessor');
+
+/**
+ * A class that processes events pushed to the data layer
+ * by constructing and sending Google Analytics events via
+ * Measurement Protocol.
+ * For its Google Analytic events, it generates a unique client ID
+ * and stores it in the long term storage model provided.
+ * @implements {EventProcessor}
+ */
+class GoogleAnalyticsEventProcessor {
+  /**
+   * @param {{
+   *     'api_secret': (string|undefined),
+   *     'measurement_id': (string|undefined),
+   *     'measurement_url': string,
+   *     'client_id_expires': number,
+   *     'automatic_params': !Object<string, boolean>,
```

Comment:
It'd be a bit more natural for users if this was an !Array<string>, and the library transforms it internally into a set (for efficiency).  It'd also enforce that the automatic_params list is additive only, which I see as a good thing.